### PR TITLE
Fix get record details endpoint

### PR DESCRIPTION
**Pull request by atatmaja**
Changelog:

- Use common list of "dashboard_source_cols" across the record_details and record endpoints
- Ensure that the returned record is always a dict rather than an array of length 1
- Convert dates to isoformat before returning
- Ignore None values in multi select fields

### CODE REVIEW

File: app/namespaces/data_provider/data_provider_service.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def _get_parsed_record(results):
     for col in results_df.columns:
         # Convert multi select values into an array
         if col in multi_select_cols:
-            multi_select_vals = list(set(getattr(results_df, col).tolist()))
+            multi_select_vals = [e for e in list(set(getattr(results_df, col).tolist())) if e is not None]
```
Comment (simonarocco) [inline app/namespaces/data_provider/data_provider_service.py:49]: can just say "if e" instead of "if e is not None"

**Review by simonarocco — COMMENTED** [review]
lgtm, nit

File: app/namespaces/data_provider/data_provider_service.py (hunk 1 above)
Comment (atatmaja) [reply app/namespaces/data_provider/data_provider_service.py:49]: Hmm don't think this should ever happen but this suggestion would behave unexpectedly if one of our multi select vals is the integer 0. 

Imo if e is not None is not too unreadable and is more explicit about what we're trying to guard against.

Comment (simonarocco) [reply app/namespaces/data_provider/data_provider_service.py:49]: sg

**Pull request merged**